src/core/raster/qgsrasterlayer.cpp

```diff
@@ -2402,6 +2404,44 @@ QString QgsRasterLayer::encodedSource( const QString &source, const QgsReadWrite
       handled = true;
     }
   }
+  else if ( providerType() == "virtualraster" )
+  {
+    //Idk which one is better, so for the moment I keep both of the or cycle
```

Comment:
My preference would be that instead of modifying `query` based on `decodedVirtualParams`, let's just modify URIs inside `decodedVirtualParams.rInputLayers` and then call `QgsRasterDataProvider::encodeVirtualRasterProviderUri()` to get updated URI - this should be more robust and keeps the details of URI encoding in one place

Reply:
Ok, I added the change in the last commit (I will delete the commented code, sorry)